Sources/SwiftKuery/Utils.swift

```diff
@@ -72,4 +72,27 @@ struct Utils {
         resultQuery += inputQuery
         return resultQuery
     }
+    
+    static func convertNamedParametersToNumbered(query: String, queryBuilder: QueryBuilder) -> (String, [String:Int]) {
+        var resultQuery = ""
+        var nameToNumber = [String:Int]()
+        var index = 1
+        let marker = queryBuilder.substitutions[QueryBuilder.QuerySubstitutionNames.numberedParameter.rawValue]
+        var inputQuery = query
+        var startRange = inputQuery.range(of: Parameter.namedParameterStart)
+        var endRange = inputQuery.range(of: Parameter.namedParameterEnd)
+        while startRange != nil && endRange != nil {
+            resultQuery += inputQuery.substring(to: startRange!.lowerBound) + marker + "\(index)"
+            let nameRange = startRange!.upperBound..<endRange!.lowerBound
+            let name = inputQuery.substring(with: nameRange)
+            nameToNumber[name] = index
+            index += 1
+            inputQuery = inputQuery.substring(from: endRange!.upperBound)
+            startRange = inputQuery.range(of: Parameter.namedParameterStart)
+            endRange = inputQuery.range(of: Parameter.namedParameterEnd)
```

Comment:
Perhaps it may be marginally better to remove `inputQuery = inputQuery.substring(from: endRange!.upperBound)` and get `startRange` and `endRange` from the original `inputQuery` using `inputQuery.range(of: xxx, range: endRange!.upperBound..<inputQuery.endIndex)` ?

Of course if this was done, `resultQuery += inputQuery.substring(to: startRange!.lowerBound)` above and `resultQuery += inputQuery` below would also need to be changed appropriately.